delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/SchemasRoutes.scala

```diff
@@ -0,0 +1,214 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, IdSegment, Label}
+import io.circe.Json
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+import monix.execution.Scheduler
+
+/**
+  * The schemas routes
+  *
+  * @param identities    the identity module
+  * @param acls          the ACLs module
+  * @param organizations the organizations module
+  * @param projects      the projects module
+  * @param schemas       the schemas module
+  */
+final class SchemasRoutes(
+    identities: Identities,
+    acls: Acls,
+    organizations: Organizations,
+    projects: Projects,
+    schemas: Schemas
+)(implicit baseUri: BaseUri, s: Scheduler, cr: RemoteContextResolution, ordering: JsonKeyOrdering)
+    extends AuthDirectives(identities, acls)
+    with CirceUnmarshalling {
+
+  import baseUri.prefixSegment
+
+  private val defaultAlias = ApiMappings.default.aliases
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractCaller { implicit caller =>
+        concat(
+          pathPrefix("schemas") {
+            concat(
+              // SSE schemas for all events
+              (pathPrefix("events") & pathEndOrSingleSlash) {
+                get {
+                  operationName(s"$prefixSegment/schemas/events") {
+                    authorizeFor(AclAddress.Root, events.read).apply {
+                      lastEventId { offset =>
+                        emit(schemas.events(offset))
+                      }
+                    }
+                  }
+                }
+              },
+              // SSE schemas for all events belonging to an organization
+              ((label | orgLabelFromUuidLookup(organizations)) & pathPrefix("events") & pathEndOrSingleSlash) { org =>
+                get {
+                  operationName(s"$prefixSegment/schemas/{org}/events") {
+                    authorizeFor(AclAddress.Organization(org), events.read).apply {
+                      lastEventId { offset =>
+                        emit(schemas.events(org, offset).leftWiden[SchemaRejection])
+                      }
+                    }
+                  }
+                }
+              },
+              // SSE schemas for all events belonging to a project
+              (projectFromRefOrUUD & pathPrefix("events") & pathEndOrSingleSlash) { ref =>
+                get {
+                  operationName(s"$prefixSegment/schemas/{org}/{project}/events") {
+                    authorizeFor(AclAddress.Project(ref), events.read).apply {
+                      lastEventId { offset =>
+                        emit(schemas.events(ref, offset).leftWiden[SchemaRejection])
+                      }
+                    }
+                  }
+                }
+              }
+            )
+          },
+          ((pathPrefix("schemas") & projectFromRefOrUUD & provide(IriSegment(shacl))) |
+            (pathPrefix("resources") & projectFromRefOrUUD & idSegment)) { (ref, schemaSegment) =>
```

Comment:
It is not simple but our routes are not simple, they are a mess, so any way this will be done won't be simple.

That said, I prefer it to be self contained. Everything happenning in schemas routes should be in schemas routes. Passing along thing to the `ResourceRoutes` it is going to end up as it was in 1.4.x where the code is difficult to follow.

What's currently in place is not too difficult to read. Maybe adding a comment is enough